src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp

```diff
@@ -0,0 +1,323 @@
+}
+
+//START_SPHINX_INCLUDE_REGISTERDATAONMESH
+void DieterichSeismicityRate::registerDataOnMesh( Group & meshBodies )
+{
+  SeismicityRateBase::registerDataOnMesh( meshBodies );
+
+  forDiscretizationOnMeshTargets( meshBodies, [&] ( string const &,
+                                                    MeshLevel & mesh,
+                                                    arrayView1d< string const > const & regionNames )
+  {
+    ElementRegionManager & elemManager = mesh.getElemManager();
+
+    elemManager.forElementSubRegions< ElementSubRegionBase >( regionNames,
+                                                              [&]( localIndex const,
+                                                                   ElementSubRegionBase & subRegion )
+    {
+      subRegion.registerField< inducedSeismicity::directEffect >( getName() );
+      subRegion.registerField< inducedSeismicity::backgroundStressingRate >( getName() );
+
+      subRegion.registerField< inducedSeismicity::logSeismicityRate >( getName() );
+      subRegion.registerField< inducedSeismicity::logSeismicityRate_n >( getName() );
+      subRegion.registerField< inducedSeismicity::logDenom >( getName() );
+      subRegion.registerField< inducedSeismicity::logDenom_n >( getName() );
+    } );
+   } );
+}
+//END_SPHINX_INCLUDE_REGISTERDATAONMESH
+
+real64 DieterichSeismicityRate::solverStep( real64 const & time_n,
+                                  real64 const & dt,
+                                  const int cycleNumber,
+                                  DomainPartition & domain )
+{
+  // Save initial stress state on pre-defined fault orienations to field variables
+  initializeMeanSolidStress(cycleNumber, domain); 
+
+  // Call member variable stress solver to update the stress state
+  real64 dtStress = m_stressSolver->solverStep(time_n, dt, cycleNumber, domain );
+
+  // Loop over subRegions to update stress on faults and solver for seismicity rate
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&]( string const &,
+                                                                 MeshLevel & mesh,
+                                                                 arrayView1d< string const > const & regionNames )
+  
+  {
+    mesh.getElemManager().forElementSubRegions( regionNames,
+                                                [&]( localIndex const,
+                                                     ElementSubRegionBase & subRegion )
+    {
+      // project new stress state to update stress on fault
+      if ( subRegion.hasWrapper( SolidMechanicsLagrangianFEM::viewKeyStruct::solidMaterialNamesString() ) )
+      {
+        updateMeanSolidStress( subRegion );
+      }
+
+      // solve for the seismcity rate given new stresses on faults
+      integralSolverStep( time_n, dtStress, subRegion ); 
+    });
+  });
+
+  // return time step size achieved by stress solver
+  return dtStress;
+}
+
+// Solve backward-Euler discretization of ODE by Newton-Raphson
```

Comment:
all this commented code should probably be removed. Or, if it works, I guess that we can clean it up and keep it in here.